src/analyses/mutexAnalysis.ml

```diff
@@ -209,7 +209,7 @@ struct
 
   let event ctx e octx =
     match e with
-    | Events.Access {exp; lvals; kind; _} when ThreadFlag.is_multi (Analyses.ask_of_ctx ctx) -> (* threadflag query in post-threadspawn ctx *)
+    | Events.Access {exp; lvals; kind; _} when ThreadFlag.has_ever_been_multi (Analyses.ask_of_ctx ctx) -> (* threadflag query in post-threadspawn ctx *)
```

Comment:
```suggestion
    | Events.Access {exp; lvals; kind; _} when ThreadFlag.is_currently_multi (Analyses.ask_of_ctx ctx) -> (* threadflag query in post-threadspawn ctx *)
```
Or is there some reason it needs to account for history?

Reply:
As above, this is needed for protection to be computed as desired.